specs/Storage.Pickers/FileSavePicker.md

```diff
@@ -0,0 +1,195 @@
+    SuggestedStartLocation = PickerLocationId.DocumentsLibrary,
+    
+    // (Optional) specify the default file name. If not specified, use system default.
+    SuggestedFileName = "My Document",
+
+    // (Optional) specify the text displayed on commit button. 
+    //     If not specified, the system uses a default label of "Open" (suitably translated).
+    CommitButtonText = "Save Document",
+
+    // (Optional) categorized extensions types. If not specified, allow All Files (*.*)
+    //     Note that when allow All Files (*.*), end users can save a file without extension.
+    FileTypeChoices = {
+        { "Documents", new List<string> { ".txt", ".doc", ".docx" } }
+    },
+
+    // (Optional) specify the default file extension (will be appended after the default file name).
+    //      If not specified, will not appended after the default extension.
+    DefaultFileExtension = ".txt",
+};
+```
+
+C++
+
+```C++
+#include <winrt/Microsoft.Windows.Storage.Pickers.h>
+using namespace winrt::Microsoft::Windows::Storage::Pickers;
+
+FileSavePicker savePicker(AppWindow().Id());
+
+// (Optional) specify the initial location.
+//     If not specified, default to PickerLocationId.Unspecified.
+savePicker.SuggestedStartLocation(PickerLocationId::DocumentsLibrary);
+
+// (Optional) specify the default file name. If not specified, use system default.
+savePicker.SuggestedFileName(L"NewDocument");
+
+// (Optional) specify the text displayed on commit button. 
+//     If not specified, the system uses a default label of "Open" (suitably translated).
+savePicker.CommitButtonText(L"Save Document");
+
+// (Optional) categorized extensions types. If not specified, allow All Files (*.*)
+//     Note that when allow All Files (*.*), end users can save a file without extension.
+savePicker.FileTypeChoices().Insert(L"Text", winrt::single_threaded_vector<winrt::hstring>({ L".txt" }));
+
+// (Optional) specify the default file extension (will be appended after the default file name).
+//      If not specified, will not appended after the default extension.
+savePicker.DefaultFileExtension(L".txt");
+```
+
+## Setting the FileSavePicker.SuggestedSaveFilePath Property
+
+The `SuggestedSaveFilePath` property of `FileSavePicker` dictates 2 key aspects of the save file 
+dialog's initial state:
+
+*   **Initial Directory:** The dialog defaults to the directory of the `SuggestedSaveFilePath`. 
+
+    This behavior overrides any picker-specific remembered folder settings and ensures the dialog 
+    opens in the suggested file's parent folder.
+
+*   **Pre-filled File Name:** The file name field in the dialog is pre-populated with the name of 
+the `SuggestedSaveFilePath`. 
+
+    This file name overrides the `FileSavePicker.SuggestedFileName` property if both are set.
+
+### Examples
+C#
+```C#
+using Microsoft.Windows.Storage.Pickers;
+
+var savePicker = new FileSavePicker(this.AppWindow.Id);
+savePicker.SuggestedSaveFilePath(@"C:\temp\MyProject\MyFile.txt");
```

Comment:
Update all C# sample snippets to use property assignment syntax, since this is no longer a method